commit ccb8e04a43425bbc4c0467d074ca004d6eeba45f (refs/changes/99/5299/2)
Author: Peter Marton <pmarton@netflix.com>
Date:   2019-01-02T14:33:52-08:00 (9 months ago)
    
    joyent/node-verror#63 do not call sprintf with one argument

diff --git a/lib/verror.js b/lib/verror.js
index 8663dde..d42a7b1 100644
--- a/lib/verror.js
+++ b/lib/verror.js
@@ -114,8 +114,15 @@ function parseConstructorArguments(args)
 		});
 	}
 
+	/*
+	 * Do not call sprintf with one argument, performance optimization.
+	 * It also addresses the issue when someone passes user input and it
+	 * contains format string characters like %s without additional args.
+	 */
 	if (sprintf_args.length === 0) {
 		shortmessage = '';
+	} else if (sprintf_args.length === 1) {
+		shortmessage = sprintf_args[0];
 	} else {
 		shortmessage = sprintf.apply(null, sprintf_args);
 	}
